tiatoolbox/visualization/tileserver_api.yml

```diff
@@ -0,0 +1,290 @@
+            type: string
+        - in: path
+          name: tile_group
+          description: Number of the TileGroup
+          required: true
+          schema:
+            type: integer
+        - in: path
+          name: z
+          description: Zoom level
+          required: true
+          schema:
+            type: integer
+        - in: path
+          name: x
+          description: X coordinate of the tile
+          required: true
+          schema:
+            type: integer
+        - in: path
+          name: y
+          description: Y coordinate of the tile
+          required: true
+          schema:
+            type: integer
+        - in: path
+          name: res
+          description: Resolution of the tile (1 or 2x)
+          required: true
+          schema:
+            type: integer
+      responses:
+        '200':
+          description: Image returned successfully
+          content:
+            image/webp:
+              schema:
+                type: image/webp
+
+  /:
+    get:
+      summary: Get index page
+      description: This endpoint serves a web page with a simple slide viewer
+      responses:
+        '200':
+          description: The index page.
+          content:
+            text/html:
+              schema:
+                type: string
+
+  /tileserver/get_user:
+    get:
+      summary: Get user information
+      description: Set up new user and generate an identifier for them
+      responses:
+        '200':
+          description: Successful response
+          headers:
+          Set-Cookie:
+            schema:
+              type: string
+            description: The 'user' cookie containing the user identifier.
+
+  /tileserver/change_color_prop/{prop}:
+    put:
+      summary: Update property to color annotations by
+      parameters:
+        - in: path
+          name: prop
+          description: Name of the property
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful response
+
+  /tileserver/change_slide/{slide_path}:
+    put:
+      summary: change the base slide. All overlays will be removed.
+      parameters:
+        - in: path
+          name: slide_path
+          description: Path to the slide in url safe form
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful response
+
+  /tileserver/change_cmap/{cmap}:
+    put:
+      summary: Change colormap for a layer
+      parameters:
+        - in: path
+          name: cmap
+          description: Name of the colormap, should be a matplotlib colormap
+            or a dict string
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful response
+
+  /tileserver/load_annotations/{file_path}/{model_mpp}:
+    put:
+      summary: Load annotations for a slide
```

Comment:
This endpoint adds annotations to the current annotation overlay layer, if there is one existing already. If one doesn't exist, then one is created. I've assigned it as 'put', because it is putting additional annotations into the annotation overlay layer (by loading them from a file).